File: Laputin/File.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Laputin
{
    public class File
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string Hash { get; set; }

        [Required]
        public string Path { get; set; }

        [Required]
        public bool Active { get; set; }

        [Required]
        public long Size { get; set; }

        [Required]
        public string Metadata { get; set; }

        [Required]
        public string Type { get; set; }

        public ICollection<Tag> Tags { get; set; }
    }
}

File: oclif/src/laputin/file.ts
import path = require('path');

import { Tag } from './tag';

export class File {
    public fileId: number;
    public hash: string;
    public path: string;
    public name: string;
    public tags: Tag[];
    public size: number;
    public metadata: any;
    public type: string;
    public active: boolean;

    constructor(
        fileId: number,
        hash: string,
        filePath: string,
        tags: Tag[],
        size: number,
        type: string,
        active: boolean,
        metadata?: any
    ) {
        this.fileId = fileId;
        this.hash = hash;
        this.path = path.normalize(filePath);
        this.name = this.cleanPath();
        this.tags = tags;
        this.size = size;
        this.type = type;
        this.active = active;
        this.metadata = metadata || {};
    }

    private cleanPath(): string {
        return path.basename(this.path);
    }
}
